code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -15,239 +152,216 @@
+	flags = CONDUCT
+	force = 15 //it's heavy
+	armour_penetration = 0
+	throwforce = 10
+	throw_speed = 1
+	throw_range = 2 //again heavy
+	w_class = WEIGHT_CLASS_BULKY //not stuffing this into a backpack
+	materials = list(MAT_METAL=500)
+	resistance_flags = FIRE_PROOF
+	origin_tech = "combat=1;plasmatech=2;engineering=2"
+	flame_intensity = 1.4
+	consumption_rate = 20
+	tank_exposure = 3
+	lit = FALSE
+	tool_behaviour = null
+	toolspeed = 0.5
+	usesound = 'sound/items/welder.ogg'
+	actions_types = list(/datum/action/item_action/toggle_flamethrowermode)
+	var/consumption_rate_welder = 5
+	var/force_welder = 30
+	var/armour_penetration_welder = 50
+	var/status = FLAMETHROWERMODE
+	var/create_with_tank = FALSE
+
+/obj/item/flamethrower/advanced/Destroy()
+	QDEL_NULL(ptank)
+	return ..()
+
+/obj/item/flamethrower/advanced/update_icon()
+	cut_overlays()
+	if(ptank)
+		add_overlay("+ptank")
+		if(ptank.air_contents.toxins > 20) //preempting that runetime.
+			add_overlay("+fuel5")
+		else if(ptank.air_contents.toxins > 15)
+			add_overlay("+fuel4")
+		else if(ptank.air_contents.toxins > 10)
+			add_overlay("+fuel3")
+		else if(ptank.air_contents.toxins > 5)
+			add_overlay("+fuel2")
+		else if(ptank.air_contents.toxins < 5)
+			add_overlay("+fuel1")
+	else 
+		add_overlay("+fuel0")
 	if(lit)
-		START_PROCESSING(SSobj, src)
-		if(!warned_admins)
-			message_admins("[ADMIN_LOOKUPFLW(user)] has lit a flamethrower.")
-			warned_admins = TRUE
+		item_state = "burner[status]"
+		add_overlay("+lit[status]")
 	else
-		STOP_PROCESSING(SSobj,src)
-	update_icon()
+		item_state = "burner0"
+		icon_state = "burner[status]"
+	if(ismob(loc))
+		var/mob/M = loc
+		M.update_inv_l_hand()
+		M.update_inv_r_hand()
 
-/obj/item/flamethrower/CheckParts(list/parts_list)
+/obj/item/flamethrower/advanced/attack(mob/living/M, mob/living/user)
+	if(user.get_inactive_hand())
+		to_chat(user, "<span class='userdanger'>You need both hands free to attack with \the [src]!</span>")
+		return
+	if(lit && status == WELDERMODE) //we want it to have different attack values depending on what status and if it's on or not
+		damtype = BURN
+		force = force_welder
+		armour_penetration = armour_penetration_welder
+		hitsound = 'sound/items/welder.ogg'
+		consume_fuel(consumption_rate)
+	else
+		damtype = BRUTE
+		force = initial(force)
+		armour_penetration = initial(armour_penetration)
+		hitsound = "swing_hit"
 	..()
-	weldtool = locate(/obj/item/weldingtool) in contents
-	igniter = locate(/obj/item/assembly/igniter) in contents
-	igniter.secured = FALSE
-	status = TRUE
-	update_icon()
 
-//Called from turf.dm turf/dblclick
-/obj/item/flamethrower/proc/flame_turf(turflist)
-	if(!lit || operating)
+/obj/item/flamethrower/advanced/afterattack(atom/target, mob/user, flag)
+	. = ..()
+	if(flag)
+		return // too close
+	if(user.get_inactive_hand())
+		to_chat(user, "<span class='userdanger'>You need both hands free to use \the [src]!</span>")
```

Comment:
Won't you get two messages now if you attack once?

Reply:
nah it only provide one of the messages, I do not know why, it might be something to do with how attack and after attack work.